src/API/auth/facebook/passport.js

```diff
@@ -1,28 +1,41 @@
-import passport from 'passport';
-import passportFacebook from 'passport-facebook';
+/* istanbul ignore next */
+// Should be ignored for now because of third party dependency. So API tests can't help
 
-const FacebookStrategy = passportFacebook.Strategy;
+import passport from 'passport';
+import FacebookTokenStrategy from 'passport-facebook-token';
 
 /**
  * @function setup
  * @description Register facebook authentication strategy
  * @param {Object} UserService
- * @param {Object} config
+ * @param {Object} facebookConfig
  */
-export default function setup(UserService, config) {
-    passport.use(new FacebookStrategy({
-        clientID: config.facebook.clientID,
-        clientSecret: config.facebook.clientSecret,
-        callbackURL: config.facebook.callbackURL
+export default function setup(UserService, facebookConfig) {
+    passport.use(new FacebookTokenStrategy({
+        clientID: facebookConfig.clientID,
+        clientSecret: facebookConfig.clientSecret
     }, (accessToken, refreshToken, profile, done) => {
-        UserService.findOrCreateUser({ email: profile.emails[0].value }, {
-            facebookId: profile.id,
-            email: profile.emails[0].value,
-            name: profile.username
-        })
+        if (profile.emails[0].value) {
+            // Some times it happens
+            // https://developers.facebook.com/bugs/298946933534016/?comment_id=620710881344692
+            // https://developers.facebook.com/docs/graph-api/reference/user
+            done(new Error('Facebook Auth: Not found e-mail in profile'));
+            return;
+        }
+
+        UserService.findOrCreateUser(
```

Comment:
Is it possible to pass only the profile as argument? Or there is other use cases expected ?